### PR TITLE
Fix difference

### LINKED ISSUE #67: Sets difference calculation is broken.
It looks like since [this commit](https://github.com/purescript/purescript-ordered-collections/commit/85ee3ba4aab99053e3225044549461d024263923) difference calculation is not working as expected.

Example:
```purescript
S.difference (S.fromFoldable [1,2,3,4]) (S.fromFoldable [2,3]) == S.empty
```

Whereas the above expression should equal to `S.fromFoldable [1,4]`
In fact, I could not find any other inputs that would not yield the empty set for this operation.

Previous versions of that library give expected results.

### PR DESCRIPTION
**Pull request by natefaubion**
**Description of the change**

Fixes `Map.Internal.difference`. Fixes #67.

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [x] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [x] Added a test for the contribution (if applicable)

### CODE REVIEW

File: test/Test/Data/Map.purs [hunk 1]
```diff
@@ -251,8 +251,12 @@ mapTests = do
   log "difference"
   quickCheck $ \(TestMap m1) (TestMap m2) ->
     let d = M.difference (m1 :: M.Map SmallKey Int) (m2 :: M.Map SmallKey String)
-    in and (map (\k -> M.member k m1) (A.fromFoldable $ M.keys d)) &&
-       and (map (\k -> not $ M.member k d) (A.fromFoldable $ M.keys m2))
+    in
+      if M.size d == 0 then
+        and (map (\k -> M.member k m2) (A.fromFoldable $ M.keys m1))
```
Comment (natefaubion) [inline test/Test/Data/Map.purs:256]: I've added an extra predicate to the test for when d is empty.

**Review by JordanMartinez — APPROVED** [review]
